Src/main.c

```diff
@@ -99,15 +100,27 @@ int main(void)
   /* USER CODE BEGIN 2 */
   debug("\n\nBootcamp starting up...");
   debug("Compiled on %s at %s", __DATE__, __TIME__);
-
+  HAL_ADC_Start(&hadc);
+  HAL_TIM_PWM_Start(&htim16,TIM_CHANNEL_1);
 
   HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+  int minDutyCycleMs = 1;
+  int maxDutyCycleMs = 2;
+  int differenceDutyCycleMs = maxDutyCycleMs - minDutyCycleMs;
+  int arrPeriod = 60000; //arrPeriod in this context is equal to the maxDutyCycleMs
+
   while (1)
   {
+    uint16_t adcOutput = 0;
+    adcOutput = atoi(HAL_ADC_GetValue(&hadc)); //This will give a value corrosponding to 0-4095 and will be between 0-3.3v. It is a 12 bit long int.
+    uint16_t adcPercentage = adcOutput/4095;
```

Comment:
Since adcOutput and adcPercentage are integers, and so is the 4095 litteral, integer math is done, and adcPercentage always evaluates to 0.

Try testing this logic on an online C compiler or something to make sure it's doing what you expect.

Reply:
Done! I realized a double makes more sense here.